src/main/java/org/opentripplanner/routing/alternativelegs/AlternativeLegs.java

```diff
@@ -256,6 +262,7 @@ private static Stream<TripPatternBetweenStops> withBoardingAlightingPositions(
       .iterate(stops.size() - 1, i -> i - 1)
       .limit(stops.size())
       .filter(i -> destinations.contains(stops.get(i)) && tripPattern.canAlight(i))
+      .limit(onlyFirstDestinationStop ? 1 : destinations.size())
```

Comment:
I think it would be enough to have `.limit(1)` here and other changes in this pr can be removed as this doesn't need to be parametrized, right?

Reply:
That is what I would have done but it seemed that the cartesian product of  all possible origin and destination stops was written intentionally without the limit so I wanted to keep that option available as well. The parametrization was removed from the LegacyGraphQL query but I'm not sure if we want to replace the original functionality with this as well

Reply:
I would guess that this behaviour of including the same trip more than once is never desired but I haven't checked what are all the use cases this code is used for. Maybe @hannesj knows if it would be fine just to have the hardcoded limit(1).

Reply:
We need to create two alternative legs for the Oslo t-bane line 5 if the trip is somewhere between Ullevål and Töyen, but only one trip per boarding.

I think it would be nice to have a unit test for behaviour on line like this.

![image](https://user-images.githubusercontent.com/698100/221803578-66a530e7-70e9-45a9-a3bf-cd60d93dba7f.png)

Reply:
Ok, so if I understood correctly, the original behavior cannot be removed with the hard limit as it is needed for generating the two alternatives at e.g. Ullevål. I think the current changes should only give one trip as I added a `distinct()` after `.filter(tripPatternPredicate)`. Otherwise, the second visit to a stop could duplicate some of the trip patterns that would be used later on in the search. Does something else still need to be done apart from adding a unit test?

Reply:
We could first gather all possible boarding positions, then generate the shortest leg from each of them, and then grouping by the alighting position and taking the shortest leg for each group.